src/t.jl

```diff
@@ -33,13 +33,13 @@ pvalue(x::TTest; tail=:both) = pvalue(TDist(x.df), x.t; tail=tail)
 default_tail(test::TTest) = :both
 
 # confidence interval by inversion
-function StatsBase.confint(x::TTest, alpha::Float64=0.05; tail=:both)
+function StatsBase.confint(x::TTest, alpha::Float64=get_alpha(x); tail=get_tail(x))
     check_alpha(alpha)
 
     if tail == :left
-        (-Inf, StatsBase.confint(x, alpha*2)[2])
+        (-Inf, StatsBase.confint(x, alpha*2, tail=:both)[2]) # tail=:both required as recursive anchor
```

Comment:
The quantiles are only computed if `tail==:both`. The recursion would be infinite if the argument `tail=:both` would not be provided because `tail` is set to the tail of `x` now, which may be one-sided.

The term "recursive anchor" is chosen because `tail==:both` will not allow infinite recursion.